telco-examples/.release_generator/generator.py

```diff
@@ -15,75 +33,243 @@ def increase_indent(self, flow=False, indentless=False):
+# Process YAML File (two-phase: mgmt then inline)
+# -----------------------------
+def process_yaml_file(filepath, release_versions, image_map):
+    try:
+        with open(filepath, "r") as f:
+            raw_text = f.read()
+    except Exception as e:
+        print(f"Skipping {filepath} — could not read: {e}")
+        return False
 
-    if not data or "kubernetes" not in data:
+    cleaned_text, placeholders = preserve_placeholders(raw_text)
+
+    try:
+        documents = list(yaml.safe_load_all(cleaned_text))
+    except yaml.YAMLError as e:
+        print(f"Skipping {filepath} — YAML error: {e}")
         return False
 
-    print(f"Updating versions in: {filepath}")
-    updated_data = update_versions_in_mgmt_file(data, release_versions)
+    changed = False
+    updated_documents = []
+
+    # Phase 1: update mgmt structure versions
+    for doc in documents:
+        if not isinstance(doc, dict):
+            updated_documents.append(doc)
+            continue
+
+        before = yaml.dump(doc, Dumper=IndentedDumper, default_flow_style=False, sort_keys=False)
+        new_doc = copy.deepcopy(doc)
+
+        update_versions_in_mgmt_file(new_doc, release_versions, image_map)
+
+        after = yaml.dump(new_doc, Dumper=IndentedDumper, default_flow_style=False, sort_keys=False)
+
+        if before != after:
+            changed = True
+
+        updated_documents.append(new_doc)
+
+    if changed:
+        output_yaml = yaml.dump_all(updated_documents, Dumper=IndentedDumper, default_flow_style=False, sort_keys=False)
+        output_yaml = restore_placeholders(output_yaml, placeholders)
+
+        with open(filepath, "w") as f:
+            f.write(output_yaml)
+        print(f"Updated mgmt versions: {filepath}")
+    else:
+        print(f"No mgmt changes: {filepath}")
+
+    # Phase 2: update inline text blocks for downstream clusters manifests
+    try:
+        with open(filepath, "r") as f:
+            updated_text = f.read()
+    except Exception as e:
+        print(f"Skipping inline update for {filepath} — could not read: {e}")
+        return changed
+
+    updated_text_with_inline = update_inline_versions_in_text(updated_text, release_versions)
+
+    if updated_text_with_inline != updated_text:
+        with open(filepath, "w") as f:
+            f.write(updated_text_with_inline)
+        print(f"Updated inline versions: {filepath}")
+        return True
+
+    print(f"No inline changes: {filepath}")
+    return changed
+
+# -----------------------------
+# Main
+# -----------------------------
+def main():
+    parser = argparse.ArgumentParser(description="Update Helm, Kubernetes, and image versions from release files")
+    parser.add_argument("version", help="Release version in X.Y.Z format (e.g., 3.3.1)")
+    parser.add_argument("--root", default="../", help="Root path to recursively scan for .yaml files")
+    parser.add_argument("--registry", choices=["internal", "public"], default="public",
```

Comment:
I would use `factory` and `production` rather than internal and public, as both are public.

Reply:
makes sense!

Reply:
changed